src/renderer/features/fellowship-referendum-details/components/ReferendumDetailsModal.tsx

```diff
@@ -21,62 +22,77 @@ export const referendumAdditionalInfoSlot = createSlot<{ referendum: Referendum
+    useGate(fellowshipReferendumsDetailsFeature.gate);
 
-  const { t } = useI18n();
+    const { t } = useI18n();
 
-  const tracks = useUnit(tracksModel.$list);
-  const evidenceContent = useUnit(details.$evidenceContent);
+    const tracks = useUnit(tracksModel.$list);
+    const evidenceContent = useUnit(details.$evidenceContent);
+    const store = useUnit(fellowship.$store);
+    const referendum = useMemo(() => {
+      return store?.referendums?.find(r => r.id === referendumId) ?? null;
+    }, [store, referendumId]);
 
-  const referendumId = referendum?.id;
-
-  const modalTitle = useMemo(() => {
-    if (title) {
-      return title;
-    }
+    const modalTitle = useMemo(() => {
+      if (title) {
+        return title;
+      }
 
-    if (referendum && referendumService.isOngoing(referendum)) {
-      if (trackService.isPromotionTrack(referendum.track) || trackService.isRetentionTrack(referendum.track)) {
-        const rankTitle = detailsService.getRankTitle(referendum.track, tracks);
-        if (rankTitle) {
-          return rankTitle;
+      if (referendum && referendumService.isOngoing(referendum)) {
+        if (trackService.isPromotionTrack(referendum.track) || trackService.isRetentionTrack(referendum.track)) {
+          const rankTitle = detailsService.getRankTitle(referendum.track, tracks);
+          if (rankTitle) {
+            return rankTitle;
+          }
         }
       }
-    }
-
-    return t('governance.referendums.referendumTitle', { index: referendumId });
-  }, [title, referendum, tracks, referendumId]);
-
-  return (
-    <Modal size="xl" height="fit">
-      <Modal.Trigger>{children}</Modal.Trigger>
-      <Modal.Title close>{modalTitle}</Modal.Title>
-      <Modal.Content>
-        <div className="flex h-full bg-main-app-background">
-          <Box direction="row" width="100%" height="100%" gap={4} padding={[4, 6]} fillContainer>
-            <Box width="100%" height="100%" gap={4}>
-              <ReferendumDescription referendum={referendum} />
-            </Box>
-            <Box width="350px" shrink={0} gap={4}>
-              <Slot id={referendumAdditionalHighPriorityInfoSlot} props={{ referendumId }} />
 
-              <MemberProfile referendum={referendum} />
+      return t('governance.referendums.referendumTitle', { index: referendumId });
+    }, [title, referendum, tracks, referendumId]);
 
-              <Slot id={referendumAdditionalInfoSlot} props={{ referendum }} />
-
-              {!isCurrentUser && <Slot id={referendumActionsSlot} props={{ referendum, evidence: evidenceContent }} />}
-
-              <AdditionalInfo referendumId={referendumId} evidenceHash={evidenceContent?.hash} />
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const handleToggle = (open: any) => {
+      if (!open && onClose) {
+        onClose();
+      }
+    };
+
+    return (
+      <Modal size="xl" height="fit" isOpen={isOpen} onToggle={onClose ? handleToggle : undefined}>
```

Comment:
not necessary, you check it inside

```suggestion
      <Modal size="xl" height="fit" isOpen={isOpen} onToggle={handleToggle}>
```

Reply:
fixed